syft/pkg/cataloger/deb/parse_dpkg_status_test.go

```diff
@@ -242,3 +259,153 @@ func TestSourceVersionExtract(t *testing.T) {
 		})
 	}
 }
+
+func Test_parseDpkgStatus(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   *bufio.Reader
+		want    []pkg.Package
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name:    "no more packages",
+			input:   bufio.NewReader(strings.NewReader(`Package: apt`)),
+			wantErr: assert.NoError,
+		},
+		{
+			name: "duplicated key",
+			input: bufio.NewReader(strings.NewReader(`
+Package: apt
+Package: apt-get
+
+`)),
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
```

Comment:
Might be slightly easier to understand by just having an error message (like `expectedError: "duplicate key discovered: Package"`) instead of a callback here, then if `expectedError != ""` in the actual test below, do this?

Reply:
I refactored the test so we still use tests but in a simpler manner, hopefully more readable this time. 
I would prefer to stick to error var, they are a simple but rich API that supports error encapsulation, and checking -- which we can't do as easily with pure strings.